endpoints/openrtb2/auction.go

```diff
@@ -1633,3 +1642,48 @@ func checkIfAppRequest(request []byte) (bool, error) {
 	}
 	return false, nil
 }
+
+func (deps *endpointDeps) ExtractFPDForBidders(req *openrtb_ext.RequestWrapper) (map[openrtb_ext.BidderName]*openrtb_ext.ORTB2, []error) {
+	errL := []error{}
+	var resolvedFPD map[openrtb_ext.BidderName]*openrtb_ext.ORTB2
+
+	//If {site,app,user}.ext.data exists, collect it and remove {site,app,user}.ext.data from request
+	globalFpdData, err := firstpartydata.ExtractGlobalFPDData(req)
+	if err != nil {
+		errL = []error{err}
+		return resolvedFPD, errL
+	}
+
+	reqExt, err := req.GetRequestExt()
+	if err != nil {
+		errL = append(errL, err)
+		return resolvedFPD, errL
+	}
+	if reqExt == nil || reqExt.GetPrebid() == nil {
+		return resolvedFPD, errL
+	}
+	biddersWithGlobalFPD := make([]string, 0)
+
+	if reqExt.GetPrebid().Data != nil {
+		biddersWithGlobalFPD = reqExt.GetPrebid().Data.Bidders
+		reqExt.GetPrebid().Data.Bidders = nil
+	}
+
+	fbdBidderConfigData, reqExtPrebid := firstpartydata.ExtractBidderConfigFPD(*reqExt.GetPrebid())
+	reqExt.SetPrebid(&reqExtPrebid)
+
+	if len(fbdBidderConfigData) == 0 && len(biddersWithGlobalFPD) == 0 {
+		return resolvedFPD, errL
+	}
+
+	//If ext.prebid.data.bidders isn't defined, the default is there's no permission filtering
+	openRtbGlobalFPD := firstpartydata.ExtractOpenRtbGlobalFPD(req.BidRequest)
+
+	var fpdErrors []error
+	resolvedFPD, fpdErrors = firstpartydata.ResolveFPDData(req.BidRequest, fbdBidderConfigData, globalFpdData, openRtbGlobalFPD, biddersWithGlobalFPD)
+	if fpdErrors != nil {
+		errL = append(errL, fpdErrors...)
+	}
+
+	return resolvedFPD, errL
```

Comment:
Since it's not possible for `errL` to be populated at this point, you can simplify this to a direct return:
```
return firstpartydata.ResolveFPDData(...)
```

Reply:
Done!